hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -292,6 +315,9 @@ protected void writeInsertRecord(HoodieRecord<T> hoodieRecord) throws IOExceptio
       return;
     }
     if (writeRecord(hoodieRecord, insertRecord, HoodieOperation.isDelete(hoodieRecord.getOperation()))) {
+      if (cdcEnabled) {
+        cdcLogger.put(hoodieRecord, null, insertRecord);
+      }
```

Comment:
Can we create a sub-class here to avoid all these `cdcEnabled` flag switching ?

Reply:
We actually can't do that w/o requiring the `insertRecord` to be deserialized twice

Reply:
What do you mean for `deserialized twice`, just overwride the `writeRecord` method and add the cdc logger logic should work here.